gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/DataFlowTopology.java

```diff
@@ -0,0 +1,150 @@
+package gobblin.data.management.copy.replication;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+
+import gobblin.data.management.copy.CopyableDataset;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * Class to represent the data flow topology from copy destination to copy sources
```

Comment:
Flows always flow from source to destination even though the actual route specification may use the destination as a key.